cpu/esp32/include/adc_arch_private.h

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2022 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp32
+ * @{
+ *
+ * @file
+ * @brief       Architecture specific internal ADC functions for ESP32
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @}
+ */
+
+#ifndef ADC_ARCH_PRIVATE_H
+#define ADC_ARCH_PRIVATE_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include "hal/adc_types.h"
+#include "periph/gpio.h"
+
+#ifndef DOXYGEN     /* hide implementation details from doxygen */
+
+#define RTCIO_GPIO(n)   n           /* n-th RTCIO GPIO */
+#define RTCIO_NA        UINT8_MAX   /* RTCIO pin not available */
+
+/**
+ * @brief   ADC hardware descriptor (for internal use only)
+ */
+typedef struct {
+    uint8_t         rtc_gpio;      /**< RTC GPIO number */
+    gpio_t          gpio;          /**< GPIO */
+    adc_unit_t      adc_ctrl;      /**< ADC controller */
+    adc_channel_t   adc_channel;   /**< channel of ADC controller */
+    char*           pad_name;      /**< symbolic name of pad */
```

Comment:
```suggestion
    gpio_t          gpio;          /**< GPIO */
    char*           pad_name;      /**< symbolic name of pad */
    uint8_t         rtc_gpio;      /**< RTC GPIO number */
    adc_unit_t      adc_ctrl;      /**< ADC controller */
    adc_channel_t   adc_channel;   /**< channel of ADC controller */    
```

We can save a few bytes of padding here

Reply:
In this case, I would prefere to keep it as is since the structure defines an entry of the map from RTCIO GPIO to GPIO which contains required information about the ADC and the channel including a pad name. That is, `rtc_gpio` can be considered as index, `gpio` is the corresponding GPIO and `adc_ctrl` and `adc_channel` are the ADC and the channel used for that RTCIO GPIO. Reordering the member would also require to change the table here https://github.com/RIOT-OS/RIOT/blob/a5067a76d35f80f7f5a66fad9fc496471a572724/cpu/esp32/periph/adc_arch.c#L40-L60.

Reply:
It seems that the change wouldn't save any byte:

- Without the change:
   ```
   text	   data	    bss	    dec	    hex	filename
  59939	  12068	   7594	  79601	  136f1	tests/periph_adc/bin/esp32-wroom-32/tests_periph_adc.elf
  ```
- With the change:
   ```
      text	   data	    bss	    dec	    hex	filename
  59939	  12068	   7594	  79601	  136f1	tests/periph_adc/bin/esp32-wroom-32/tests_periph_adc.elf
   ```

Reply:
Just go ahead then